homeassistant/components/flickelectric/config_flow.py

```diff
@@ -0,0 +1,81 @@
+"""Config Flow for Flick Electric integration."""
+from asyncio import TimeoutError
+import logging
+
+import async_timeout
+from pyflick.authentication import AuthException, SimpleFlickAuth
+from pyflick.const import DEFAULT_CLIENT_ID, DEFAULT_CLIENT_SECRET
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+)
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Optional(CONF_CLIENT_ID): str,
+        vol.Optional(CONF_CLIENT_SECRET): str,
+    }
+)
+
+
+class FlickConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Flick config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def _validate_input(self, user_input):
+        auth = SimpleFlickAuth(
```

Comment:
What kind of auth is this? Is our oauth2 helper relevant?
https://developers.home-assistant.io/docs/config_entries_config_flow_handler#configuration-via-oauth2

Reply:
The API is OAuth with grant type password (ie no redirect, just send the username/password to the token endpoint).

The code for that is here: https://github.com/ZephireNZ/PyFlick/blob/3c4fae7380bcfcea9b1e0cb02444a07626261998/pyflick/authentication.py#L31-L55

I initially tried to use the OAuth scaffold but it seems like it's designed more for the redirect-and-login type of authentication.

Reply:
Yeah, that's correct. :+1:

Reply:
OK cool, no  changes needed in that case for the auth flow?

Reply:
No, I don't think our helper is adapted to that flow.  CC @balloob